app/pb/communities.proto

```diff
@@ -0,0 +1,199 @@
+syntax = "proto3";
+
+package org.couchers.api.communities;
+
+import "google/protobuf/timestamp.proto";
+
+import "pb/groups.proto";
+import "pb/pages.proto";
+
+service Communities {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  rpc GetCommunity(GetCommunityReq) returns (Community) {
+    // Get info about a Community
+  }
+
+  rpc ListCommunities(ListCommunitiesReq) returns (ListCommunitiesRes) {
+    // Gets a paginated list of sub-communities associated with a community
+  }
+
+  rpc ListGroups(ListGroupsReq) returns (ListGroupsRes) {
+    // Gets a paginated list of groups associated with a community
+  }
+
+  rpc ListAdmins(ListAdminsReq) returns (ListAdminsRes) {
+    // Gets a paginated list of community admins only
+  }
+
+  rpc ListMembers(ListMembersReq) returns (ListMembersRes) {
+    // Gets a paginated list of community members, including admins
+  }
+
+  rpc ListNearbyUsers(ListNearbyUsersReq) returns (ListNearbyUsersRes) {
+    // Gets a paginated list of users who live within the polygon of the community
+  }
+
+  rpc ListPlaces(ListPlacesReq) returns (ListPlacesRes) {
+    // Returns a paginated list of places associated with a community
+  }
+
+  rpc ListGuides(ListGuidesReq) returns (ListGuidesRes) {
+    // Returns a paginated list of guides associated with a community
+  }
+
+  rpc ListEvents(ListEventsReq) returns (ListEventsRes) {
+    // Returns a paginated events of guides associated with a community
+  }
+
+  rpc ListDiscussions(ListDiscussionsReq) returns (ListDiscussionsRes) {
+    // Returns a paginated list of discussions associated with a community
+  }
+}
+
+message Community {
+  int64 community_id = 1;
+  string name = 2;
+  string slug = 3;
```

Comment:
Done. I'd prefer to document it in a more central location as these will eventually be generated for basically every entity with a URL, but I've for now added the same comment to each `slug` field.

Reply:
Thanks. Since the slugify library is available in javascript as well, can't we leave it up to the frontends to generate the slug? Not a big deal though.